### DISCUSSION TITLE
error while updating mamba

### DISCUSSION THREAD

**Discussion by harveyzhengxh** [idaholab/moose#24971] [Q&A Getting Started] [answered]
Hi
I installed moose in 2022/04 and updated it first time yesterday.
after Update MOOSE repository by:
`cd ~/projects/moose`
`git fetch origin`
`git rebase origin/master`

I check the website [https://mooseframework.inl.gov/getting_started/new_users.html](url) and update by mamba:
`mamba activate moose`
`mamba update --all`

but the following error comes up while compiling my app:
`/home/nsi/projects/moose/framework/build/header_symlinks/MathFVUtils.h:196:44: error: 'TensorTraits' is not a member of 'libMesh::TensorTools'
  196 |   else if constexpr (libMesh::TensorTools::TensorTraits<T1>::rank == 1)
      |                                            ^~~~~~~~~~~~
/home/nsi/projects/moose/framework/build/header_symlinks/MathFVUtils.h:196:59: error: expected primary-expression before '>' token
  196 |   else if constexpr (libMesh::TensorTools::TensorTraits<T1>::rank == 1)
      |                                                           ^
/home/nsi/projects/moose/framework/build/header_symlinks/MathFVUtils.h:196:62: error: '::rank' has not been declared; did you mean 'std::rank'?
  196 |   else if constexpr (libMesh::TensorTools::TensorTraits<T1>::rank == 1)
      |                                                              ^~~~
      |                                                              std::rank
`
Then I update conda by:
`conda activate moose`
`conda update --all`
which  didn't solve the error

when I execute `mamba update --all` again comes the error:
`Traceback (most recent call last):
  File "/home/nsi/mambaforge3/bin/mamba", line 7, in <module>
    from mamba.mamba import main
  File "/home/nsi/mambaforge3/lib/python3.9/site-packages/mamba/mamba.py", line 51, in <module>
    from mamba import repoquery as repoquery_api
  File "/home/nsi/mambaforge3/lib/python3.9/site-packages/mamba/repoquery.py", line 12, in <module>
    def _repoquery(query_type, q, pool, fmt=api.QueryFormat.JSON):
AttributeError: module 'libmambapy' has no attribute 'QueryFormat'
`
the error also comes when executing `mamba env remove -n moose` and `mamba init`

I am not sure whether the error has something to do with "conda update --all" or not and I don't know how to deal with the error, which makes me fail to compile my app and moose/test.
Thanks very much

**Comment by GiudGiud** [comment, marked as answer]
Hello

if mamba update fails, this probably means the environment is borked. 

Delete the moose mamba environment then re install it following the original instructions getting started with moose on Linux/mac

Guillaume

**Reply by milljm** [reply]
Well @harveyzhengxh is trying to delete the moose environment `mamba env remove -n moose` but that also causes the error. When this happens, the correct thing to do, would be to reinstall Mambaforge all together.

```bash
rm -rf /home/nsi/mambaforge3
```

Close any opened terminals, open them again, and follow through the installation procedures from the beginning. If the problem persists, this could mean there is something "python" related in your bash or other shell profiles sourcing "something" which interferes with Mambaforge "somehow". We can try to diagnose that, once you've reinstalled Mambaforge, and if you do get the same errors...

**Reply by harveyzhengxh** [reply]
thanks for your answer,i solve the error by reinstall mamba first. 
`conda install -c conda-forge mamba`
then reinstall the moose mamba environment by:
`mamba env remove -n moose`
`mamba create -n moose moose-dev`
I am capable of compiling moose and my own app now
